src/topwidget.cpp

```diff
@@ -94,3 +98,33 @@ void TopWidget::mouseMoveEvent(QMouseEvent *event) {
     parentWidget()->move(delta);
     event->accept();
 }
+
+QToolButton* TopWidget::getBackButton() const
+{
+    auto app = KiwixApp::instance();
+    QAction *back = app->getAction(KiwixApp::HistoryBackAction);
+    return qobject_cast<QToolButton*>(widgetForAction(back));
+}
+
+QToolButton* TopWidget::getForwardButton() const
+{
+    auto app = KiwixApp::instance();
+    QAction *forward = app->getAction(KiwixApp::HistoryForwardAction);
+    return qobject_cast<QToolButton*>(widgetForAction(forward));
+}
+
+void TopWidget::updateBackForwardButtons()
+{
+    WebView *webview = KiwixApp::instance()->getTabWidget()->currentWebView();
+
+    if (webview) {
+        back_menu.reset(webview->getHistoryBackMenu());
+        forward_menu.reset(webview->getHistoryForwardMenu());
+    } else {
+        back_menu.reset();
+        forward_menu.reset();
+    }
+
+    getBackButton()->setMenu(back_menu.get());
```

Comment:
The get method has been introduce in qt 5.11 and bionic use qt 5.9 (https://doc.qt.io/qt-5/qscopedpointer.html#get)
This is why the CI doesn't compile on bionic.

Reply:
Fixed, used data()